### PR TITLE
Replace Suggesters with namedObject

### PR DESCRIPTION
**Pull request by nik9000**
Removes another parser registery type thing in favor of
`XContentParser#namedObject`. Yet another parser registry
consolidated!

Breaks that plugin API for suggesters by making defining an
extra suggester more like anything else.

### CODE REVIEW

**Comment by nik9000** [comment]
@javanna, would you like to have a look at this? More fun!

File: core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestionBuilder.java [hunk 1]
```diff
@@ -67,27 +65,26 @@
      *     "payload" : STRING_ARRAY
      * }
      */
-    private static ObjectParser<CompletionSuggestionBuilder.InnerBuilder, ParseFieldMatcherSupplier> TLP_PARSER =
-        new ObjectParser<>(SUGGESTION_NAME, null);
+    private static final ObjectParser<CompletionSuggestionBuilder.InnerBuilder, Void> PARSER = new ObjectParser<>(SUGGESTION_NAME, null);
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestionBuilder.java]: thanks for the help here too <3

**Review by javanna — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
Thanks again for reviewing @javanna!

master: 3fb9254b9548d3514c3d187081bfebfce3a948f8
5.x:

**Pull request merged**